programs/uxd/src/instructions/rebalance_alloyx_vault_depository.rs

```diff
@@ -141,14 +142,70 @@ pub struct RebalanceAlloyxVaultDepository<'info> {
 }
 
 pub(crate) fn handler(ctx: Context<RebalanceAlloyxVaultDepository>, vault_id: &str) -> Result<()> {
-    let redeemable_amount_under_management = ctx
+    // ---------------------------------------------------------------------
+    // -- Phase 1
+    // -- Compute the targets and redeemable amounts,
+    // -- We can then decide if we want to withdraw or deposit
+    // ---------------------------------------------------------------------
+
+    let router_depositories_target_redeemable_amount =
+        calculate_router_depositories_target_redeemable_amount(
+            &ctx.accounts.controller,
+            &ctx.accounts.identity_depository,
+            &ctx.accounts.mercurial_vault_depository,
+            &ctx.accounts.credix_lp_depository,
+            &ctx.accounts.alloyx_vault_depository,
+        )?;
+
+    let identity_depository_target_redeemable_amount =
+        router_depositories_target_redeemable_amount.identity_depository_target_redeemable_amount;
+    let alloyx_vault_depository_target_redeemable_amount =
+        router_depositories_target_redeemable_amount
+            .alloyx_vault_depository_target_redeemable_amount;
+
+    let identity_depository_redeemable_amount_under_management = checked_as_u64(
+        ctx.accounts
+            .identity_depository
+            .load()?
+            .redeemable_amount_under_management,
+    )?;
+    let alloyx_vault_depository_redeemable_amount_under_management = ctx
         .accounts
         .alloyx_vault_depository
         .load()?
         .redeemable_amount_under_management;
 
+    let identity_depository_underflow_value = identity_depository_target_redeemable_amount
+        .saturating_sub(identity_depository_redeemable_amount_under_management);
+    let alloyx_vault_depository_underflow_value = alloyx_vault_depository_target_redeemable_amount
+        .saturating_sub(alloyx_vault_depository_redeemable_amount_under_management);
+
+    let identity_depository_overflow_value = identity_depository_redeemable_amount_under_management
+        .saturating_sub(identity_depository_target_redeemable_amount);
+    let alloyx_vault_depository_overflow_value =
+        alloyx_vault_depository_redeemable_amount_under_management
+            .saturating_sub(alloyx_vault_depository_target_redeemable_amount);
+
+    let deposit_redeemable_amount = std::cmp::min(
+        identity_depository_overflow_value,
+        alloyx_vault_depository_underflow_value,
+    );
+    let withdraw_redeemable_amount = std::cmp::max(
```

Comment:
If the values are equal in some scenario, what happen, does the order matter

Reply:
If there is somehow a conflict, the deposited value will be used, then the withdraw value will not be used (a single rebalancing will only do a single withdraw/deposit, not both)

In all cases however, one of the two value will always be zero:
- if the overflow value > 0, then the underflow value == 0
- So there is always 2 values set to zero in the min/max expressions
- So there is always one of the two deposit/withdraw value that is zero